ssz/src/test/java/tech/pegasys/teku/ssz/backing/SszCompositeAbstractTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.ssz.backing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import tech.pegasys.teku.ssz.backing.schema.SszCompositeSchema;
+import tech.pegasys.teku.ssz.backing.schema.SszSchema;
+
+public interface SszCompositeAbstractTest extends SszDataAbstractTest {
```

Comment:
`interface` is kind of weird here - I guess it works but I've only ever seen abstract classes used for this kind of base class.

Typically they're not named ending in `Test` but something like `TestBase` since they don't run themselves, just extended by classes that do run tests.

Reply:
I can see some multiple inheritance going on in tests which is likely why these are interfaces.  I'd probably still rename them to `*TestBase` rather than just `*Test` though.

Reply:
TBH I found this approach pretty cool. I.e. you may write test cases for individual fine grained interfaces and then combine them for instances which implement one or more of those interfaces. 

Our case example:
```java
class SszBitvectorTest implements SszPrimitiveCollectionAbstractTest, SszVectorAbstractTest
```
I.e. we have a number tests which are specific for `SszBitvector` and they are in `SszBitvectorTest` but we also want to test it's generic parent interfaces `SszPrimitiveCollection` and `SszVector`. Thus the test implements corresponding test interfaces. 
But since our `SszBitvector` implementation doesn't support mutable version the final test doesn't implement `SszMutablePrimitiveCollectionAbstractTest`

BTW here is the first link I just googled with the artictle describing this approach: https://keyholesoftware.com/2018/02/06/encouraging-good-behavior-with-junit-5-test-interfaces/

Reply:
I named them `*AbstractTest` but no problems with sticking to `*TestBase` pattern if that is our general style.

Reply:
Renamed classes: b1d5082